### ISSUE TITLE
Boxed traits should work in unions.

### ISSUE THREAD

**Issue by boardmaster357** [async-graphql/async-graphql#780]
Support for graphql trait objects has been added, but it should also work as a member of a Union

The following fails with the error: the `trait bound Box<dyn ProductTrait>: async_graphql::ObjectType is not satisfied`

```
pub trait ProductTrait: Send + Sync {
    fn id(&self) -> &str;
}

#[Object]
impl dyn ProductTrait {
    #[graphql(name = "id")]
    async fn gql_id(&self, _ctx: &Context<'_>) -> &str {
        self.id()
    }
}

#[derive(Union)]
pub enum Content {
    Product(Box<dyn ProductTrait>),
}
```

**Comment by sunli829** [comment]
Thanks for the sponsorship, I'll fixed this tomorrow. 🙂

**Comment by sunli829** [comment]
Fixed in `3.0.21`
